Sources/PubNub/Networking/Endpoints/SpaceObjectsEndpoint.swift

```diff
@@ -0,0 +1,252 @@
+//
+//  ObjectsEndpoint.swift
+//
+//  PubNub Real-time Cloud-Hosted Push API and Push Notification Client Frameworks
+//  Copyright © 2019 PubNub Inc.
+//  https://www.pubnub.com/
+//  https://www.pubnub.com/terms
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+// MARK: - Space Protocols
+
+public protocol PubNubSpace: ObjectIdentifiable {
+  var name: String { get }
+  var spaceDescription: String? { get }
+}
+
+extension PubNubSpace {
+  var isValid: Bool {
+    return !id.isEmpty && !name.isEmpty
+  }
+
+  var spaceObject: SpaceObject {
+    guard let object = self as? SpaceObject else {
+      return SpaceObject(self)
+    }
+    return object
+  }
+}
+
+extension PubNubSpace where Self: Equatable {
+  func isEqual(_ other: PubNubSpace?) -> Bool {
+    return id == other?.id &&
+      name == other?.name &&
+      spaceDescription == other?.spaceDescription &&
+      custom.allSatisfy { other?.custom[$0]?.scalarValue == $1.scalarValue }
+  }
+}
+
+public protocol UpdatableSpace: PubNubSpace {
+  var updated: Date { get }
+  var eTag: String { get }
+}
+
+extension UpdatableSpace {
+  var spaceObject: SpaceObject {
+    guard let object = self as? SpaceObject else {
+      return SpaceObject(updatable: self)
+    }
+    return object
+  }
+}
+
+// MARK: - Response Decoder
+
+struct SpaceObjectResponseDecoder: ResponseDecoder {
+  typealias Payload = SpaceObjectResponsePayload
+}
+
+struct SpaceObjectsResponseDecoder: ResponseDecoder {
+  typealias Payload = SpaceObjectsResponsePayload
+}
+
+// MARK: - Membership Response Decoder
+
+struct SpaceMembershipsObjectsResponseDecoder: ResponseDecoder {
+  typealias Payload = SpaceMembershipResponsePayload
```

Comment:
Why is the `struct` plural "Memberships" but the Response Payload singular "Membership" ?

Reply:
Yes it should just be a single.